spring-components/rdf4j-spring/src/main/java/org/eclipse/rdf4j/sparqlbuilder/constraint/propertypath/InversePredicatePath.java

```diff
@@ -0,0 +1,16 @@
+package org.eclipse.rdf4j.sparqlbuilder.constraint.propertypath;
+
+import org.eclipse.rdf4j.sparqlbuilder.rdf.Iri;
+
+public class InversePredicatePath implements PredicatePathOrInversePredicatePath {
```

Comment:
Ah, I didn't realize there was lot more SparqlBuilder stuff. We'll need some time to go over this and figure out which parts are new and which overlap with existing functionality.

Might perhaps be a good idea to take these out of this PR and put them up separately? It's definitely going to be useful but it's not really related to the spring component, so I wouldn't want it to hold that up.

Reply:
Ok, I'll make a separate PR with just the SparqlBuilder stuff

Reply:
On second thought: there is quite some use of the `ExtendedVariable` class in the spring-related code, so at least in that regard, the spring code depends on the sparqlbuilder code. How do you suggest we handle that? 

The only point of the `ExtendedVariable` is to get public access to the variable name, maybe we can just modify `Variable` to provide that?

Reply:
I am keeping the code around for the time being